### PR TITLE
Initial commit for adding sle hpc distro

### PR DESCRIPTION
**Pull request by jamesongithub**
- adapt slurm install for sle hpc distro
- update job submit plugin compilation for configurable distro
- add configurable launch parameters for slurm
- add missing db install and init for slurm accounting

### CODE REVIEW

File: project.ini [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
...rpm, slurm-example-configs-20.11.7-1.el7.x86_64.rpm, slurm-openlava-20.11.7-1.el7.x86_64.rpm, slurm-pam_slurm-20.11.7-1.el7.x86_64.rpm, slurm-perlapi-20.11.7-1.el7.x86_64.rpm, slurm-slurmctld-20.11.7-1.el7.x86_64.rpm, slurm-slurmd-20.11.7-1.el7.x86_64.rpm, slurm-slurmdbd-20.11.7-1.el7.x86_64.rpm, slurm-torque-20.11.7-1.el7.x86_64.rpm, slurm_20.11.7-1_amd64.deb, slurm-pam-slurm_20.11.7-1_amd64.deb, slurm-contribs_20.11.7-1_amd64.deb, slurm-perlapi_20.11.7-1_amd64.deb, slurm-devel_20.11.7-1_amd64.deb, slurm-slurmctld_20.11.7-1_amd64.deb, slurm-example-configs_20.11.7-1_amd64.deb, slurm-slurmd_20.11.7-1_amd64.deb, slurm-libpmi_20.11.7-1_amd64.deb, slurm-slurmdbd_20.11.7-1_amd64.deb, slurm-openlava_20.11.7-1_amd64.deb, slurm-torque_20.11.7-1_amd64.deb, slurm-20.11.7-1.el8.x86_64.rpm, slurm-contribs-20.11.7-1.el8.x86_64.rpm, slurm-devel-20.11.7-1.el8.x86_64.rpm, slurm-example-configs-20.11.7-1.el8.x86_64.rpm, slurm-openlava-20.11.7-1.el8.x86_64.rpm, slurm-pam_slurm-20.11.7-1.el8.x86_64.rpm, slurm-perlapi-20.11.7-1.el8.x86_64.rpm, slurm-slurmctld-20.11.7-1.el8.x86_64.rpm, slurm-slurmd-20.11.7-1.el8.x86_64.rpm, slurm-slurmdbd-20.11.7-1.el8.x86_64.rpm, slurm-torque-20.11.7-1.el8.x86_64.rpm
```
Comment (jamesongithub) [inline project.ini:8]: diff is hard to read.
one additional file `job_submit_cyclecloud_suse_20.11.9-1.so` 
inserted at fifth position

File: specs/default/chef/site-cookbooks/slurm/attributes/default.rb [hunk 2]
```diff
@@ -40,3 +25,24 @@
 default[:slurm][:accounting][:url] = 'localhost'
 
 default[:slurm][:ha_enabled] = false
+default[:slurm][:launch_parameters] = ''
+
+myplatform=node[:platform_family]
```
Comment (jamesongithub) [inline specs/default/chef/site-cookbooks/slurm/attributes/default.rb:30]: allow overrides for distros after defaults set

File: specs/default/chef/site-cookbooks/slurm/recipes/accounting.rb [hunk 3]
```diff
@@ -60,13 +60,42 @@
...
+             EOH
+    end
 end
 
 # Configure slurmdbd.conf
 template '/sched/slurmdbd.conf' do
     owner "#{slurmuser}"
     source "slurmdbd.conf.erb"
-    action :create_if_missing
+    action :create
```
Comment (jamesongithub) [inline specs/default/chef/site-cookbooks/slurm/recipes/accounting.rb:98]: ok to allow updates to slurmdbd.conf?

**Comment by jamesongithub** [comment]
@anhoward did not have permissions to assign. please review. thanks!

**Comment by anhoward** [comment]
> @anhoward did not have permissions to assign. please review. thanks!

I'm reviewing this today and hope to merge it in this afternoon.

**Comment by jamesongithub** [comment]
@anhoward i guess this was merged via rebased branch? 

closing this one. lmk if otherwise

**Pull request closed without merging**